cmd/alertmanager/main.go

```diff
@@ -388,6 +396,45 @@ func (s peerDescSlice) Len() int           { return len(s) }
 func (s peerDescSlice) Less(i, j int) bool { return s[i].UID < s[j].UID }
 func (s peerDescSlice) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
 
+// meshSettle set isReady to 1 once the mesh is considered settle (or after a
+// configurable timeout). The idea is that we don't want to start "working" before
+// we get a chance to know most of the alerts and/or silences.
+func meshSettle(r *mesh.Router, isReady *int32, timeout time.Duration, logger log.Logger) {
+	// Inspired from https://github.com/apache/cassandra/blob/7a40abb6a5108688fb1b10c375bb751cbb782ea4/src/java/org/apache/cassandra/gms/Gossiper.java
+	// This is clearly not perfect or strictly correct but should prevent the alertmanager
+	// to be used before it is obviously not read.
+	const PollInterval = 1 * time.Second
+	const NumOkayRequired = 3
+	const ForceAfter = 60 // Limit this to 10 * PollInterval, this needs to be higher than the mesh connection timeout.
+	level.Info(logger).Log("msg", "Waiting for gossip to settle...")
+	time.Sleep(PollInterval)
+	
+	nPeers := 0
+	nOkay := 0
+	totalPolls := 0
+	for {
+		time.Sleep(PollInterval)
+		if nOkay >= NumOkayRequired {
+			level.Info(logger).Log("msg", "No gossip backlog; proceeding")
```

Comment:
Not sure about this comment. Shouldn't it be `"gossip settled, proceeding"`?

Reply:
done